pkg/machinery/client/config/path.go

```diff
@@ -27,6 +27,17 @@ func GetDefaultPath() (string, error) {
 		return path, nil
 	}
 
+	talosSAPath := filepath.Join(constants.ServiceAccountMountPath, constants.ServiceAccountTalosconfigFilename)
+
+	_, err := os.Stat(talosSAPath)
+	if err != nil && !os.IsNotExist(err) {
+		return "", err
+	}
+
+	if err == nil {
```

Comment:
isn't this redundant as the previous if block handle it? :thinking:

Reply:
Previous block only handles it if it is something else than `NotExist` error. Here, it can still be `NotExist`. If it exists, we return. Otherwise we let it fallback to `~/.talos/config`.

Reply:
ahh I missed that completely :sweat_smile: :+1: